bibcat/tests/core/test_paper.py

```diff
@@ -286,6 +286,566 @@ def test_check_truematch__variety1(self):
 
                     self.assertEqual(test_res["bool"], answer)
 
+    # For tests of _early_false_no_keyword_match:
+    if True:
```

Comment:
I don't think you're suppose to wrap tests in `if True` blocks anymore, in modern code practices.  

When writing new tests or converting old tests,  I think we should migrate the tests to `pytest` where possible.  Skimming through these tests, I think a lot here can be converted into smaller parametrized tests, with fixtures for any repeatable bits.   That would clean up these tests a bit and get rid of all the for loops.  We might not need a top level `TestPaper(unittest.TestCase)` class anymore.